code/parachain/frame/pallet-multihop-xcm-ibc/src/lib.rs

```diff
@@ -385,24 +451,27 @@ pub mod pallet {
 				},
 				_ => {
 					//emit event
+					// log error, return error, not event
 					<Pallet<T>>::deposit_event(crate::Event::<T>::FailedMatchLocation {});
 					return None
 				},
 			};
 
 			let (pallet_id, address_from, route_id, mut addresses) = location_info;
 
+			/// what if PalletInstanceId is const, just asking, can you match on it?
 			if *pallet_id != T::PalletInstanceId::get() {
 				<Pallet<T>>::deposit_event(crate::Event::<T>::FailedCallback {
```

Comment:
I think `Self` should work and is easier to write:

```suggestion
				Self::deposit_event(crate::Event::<T>::FailedCallback {
```